slyd/slyd/splash/ferry.py

```diff
@@ -209,22 +221,32 @@ def user(self):
 
     def onConnect(self, request):
         try:
-            request.auth_info = json.loads(request.headers['x-auth-info'])
+            auth_info = json.loads(request.headers['x-auth-info'])
         except (KeyError, TypeError):
             return
         self.start_time = monotonic()
         self.spent_time = 0
         self.session_id = ''
-        self.factory[self] = User(request.auth_info)
+        self.auth_info = auth_info
+        self.factory[self] = User(auth_info)
 
     def onOpen(self):
         if self not in self.factory:
             self.sendClose(1000, 'Invalid Connection missing required '
                                  'parameters')
 
     def onMessage(self, payload, isbinary):
+        pool = getattr(Repoman, 'pool', None)
         payload = payload.decode('utf-8')
         data = json.loads(payload)
+        project = data.get('project', data.get('_meta', {}).get('project'))
+        self.manager = self.spec_manager.project_spec(project, self.auth_info)
+        self.manager.pm = self.spec_manager.project_manager(self.auth_info)
+        if pool is None:
+            return wrap_message_callback(self, data)
+        return pool._runWithConnection(wrap_message_callback, self, data)
```

Comment:
wrap_message_callback takes a connection as the first argument
also runWithConnection should be used instead of _runWithConnection to not block the thread.

i propose that wrap_message_callback just return the response message, then you can change the above to
```python
params = []
if pool is not None:
    params.append(pool.runWithConnection)
params.extend([wrap_message_callback, connection, self, data])
deferred = twisted.internet.defer.maybeDeferred(*params)
deferred.addCallback(self.sendMessage)
```

the wrap_callback from gitstorage/projects could be reused here instead of wrap_message_callback